.github/workflows/ci.yml

```diff
@@ -143,54 +143,28 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v3
-
       - name: Get branch name
         id: branch
         # strip `refs/heads/` from $GITHUB_REF and replace `/` with `-` so that
         # it can be used as a docker tag
         run: echo "branch=$(echo ${GITHUB_REF#refs/heads/} | tr / -)" >> "$GITHUB_OUTPUT"
 
-      # We are only using ghcr here for CI as `setup-gcloud` is a bit slow
-      # Should revisit this when we move off of google cloud build (we may want to move these to GCR)
-      - name: Registry login
+      - uses: getsentry/action-build-and-push-images@a53f146fc1ea3cb404f2dcf7378f5b60dd98d3ca
+        with:
+          image_name: 'snuba-ci'
+          platforms: 'linux/amd64'
+          dockerfile_path: 'Dockerfile'
+          ghcr: false
+          tag_nightly: false
+          tag_latest: false
+          tags: 'ghcr.io/getsentry/snuba-ci:${{ github.event.pull_request.head.sha || github.sha }}'
+          outputs: 'type=docker,dest=/tmp/snuba-ci.tar'
+
+      - name: Push snuba-ci image to GHCR
         run: |
           echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io -u $GITHUB_ACTOR --password-stdin
-
-      - run: docker buildx create --driver docker-container --use
-
-      - name: Build snuba docker image for CI
-        run: |
-          set -euxo pipefail
-
-          args=()
-
-          if [ ${{ github.event_name }} = 'push' ]; then
-            args+=(
-              --tag ghcr.io/getsentry/snuba-ci:latest
-              --cache-to type=registry,ref=ghcr.io/getsentry/snuba-ci:latest,mode=max
-              --push
-            )
-          fi
-
-          if [ ${{ github.event.pull_request.head.repo.full_name }} = ${{ github.repository }} ]; then
-            args+=(
-              --cache-to type=registry,ref=ghcr.io/getsentry/snuba-ci:${{ steps.branch.outputs.branch }},mode=max
-            )
-          fi
-
-          docker buildx build \
-            --pull \
-            --cache-from type=registry,ref=ghcr.io/getsentry/snuba-ci:${{ github.event.pull_request.head.sha || github.sha }} \
-            --cache-from type=registry,ref=ghcr.io/getsentry/snuba-ci:${{ steps.branch.outputs.branch }} \
-            --cache-from type=registry,ref=ghcr.io/getsentry/snuba-ci:latest \
-            --tag ghcr.io/getsentry/snuba-ci:${{ github.event.pull_request.head.sha || github.sha }} \
-            --tag ghcr.io/getsentry/snuba-ci:${{ steps.branch.outputs.branch }} \
-            --target testing \
-            "${args[@]}" \
-            -o type=docker,dest=/tmp/snuba-ci.tar \
-            .
+          docker load --input /tmp/snuba-ci.tar
```

Comment:
Unfortunately this can't be combined together with the `action-build-and-push-images` action because of annotations that are added

Reply:
The `latest` tag doesn't seem like it is too useful for snuba-ci and does not appear to be used anywhere, I removed it but let me know if it is worth keeping around.